rules/key_processor.go

```diff
@@ -117,9 +119,15 @@ func (v3kp *v3KeyProcessor) processKey(key string, value *string, api readAPI, l
 	v3kp.kpChannel <- task
 }
 
-func (v3kp *v3KeyProcessor) bufferCapacitySampler() {
+func (v3kp *v3KeyProcessor) bufferCapacitySampler(logger *zap.Logger) {
 	for {
-		metrics.KeyProcessBufferCap(cap(v3kp.kpChannel) - len(v3kp.kpChannel))
+		remainingBuffer := cap(v3kp.kpChannel) - len(v3kp.kpChannel)
+		metrics.KeyProcessBufferCap(remainingBuffer)
+		currentHour := time.Now().UTC().Hour()
+		if (float32(remainingBuffer)/float32(cap(v3kp.kpChannel))) < 0.05 && v3kp.lastNotified != currentHour {
```

Comment:
Should we log the remaining buffer regardless and then in LogDNA decide the threshold for the alert?

Reply:
Do you mean to log every minute and then set the logdna alert to `if >10 occurrences in past 30m` ?